pkg/tools/bash_tool.go

```diff
@@ -36,12 +37,13 @@ const (
 // expandShellVar expands shell variables and syntax using bash
 func expandShellVar(value string) (string, error) {
 	if strings.Contains(value, "~") {
-		cmd := exec.Command(bashBin, "-c", fmt.Sprintf("echo %s", value))
-		output, err := cmd.Output()
-		if err != nil {
-			return "", err
+		if len(value) >= 2 && value[0] == '~' && os.IsPathSeparator(value[1]) {
+			if runtime.GOOS == "windows" {
+				value = filepath.Join(os.Getenv("USERPROFILE"), value[2:])
+			} else {
+				value = filepath.Join(os.Getenv("HOME"), value[2:])
+			}
 		}
-		return strings.TrimSpace(string(output)), nil
```

Comment:
It may be better to prioritize the expansion of environment variables over the expansion of `~`. What do you think?